src/jobsautoreport/report.py

```diff
@@ -1,31 +1,151 @@
-        return len(jobs_list)
+        job_list = self._querier.query_e2e_jobs_by_job_type_and_state(
+            from_date, to_date, job_type, job_state
+        )
+        return len(job_list)
 
-    def get_average_duration_of_jobs(
-        self, from_date: datetime, to_date: datetime
-    ) -> Optional[timedelta]:
-        jobs_list = self._querier.query_number_of_jobs_triggered(from_date, to_date)
-        if len(jobs_list) == 0:
-            return None
-        avg = int(sum([job.duration for job in jobs_list]) / len(jobs_list))
-        return timedelta(seconds=avg)
+    def _get_success_rate_for_e2e_jobs_by_type(
+        self, from_date: datetime, to_date: datetime, job_type: JobType
+    ) -> float:
+        total_jobs = self._get_number_of_e2e_jobs_by_type(from_date, to_date, job_type)
+        successful_jobs = self._get_number_of_e2e_jobs_by_type_and_state(
+            from_date, to_date, job_type, JobState.SUCCESS
+        )
+        if total_jobs == 0:
+            return 0
+        return float(round((successful_jobs / total_jobs) * 100, 2))
+
+    def _get_top_n_e2e_jobs_by_type_and_state(
+        self,
+        from_date: datetime,
+        to_date: datetime,
+        job_type: JobType,
+        job_state: JobState,
+        n: int,
+    ) -> list[tuple[str, float]]:
+        res = []
+        jobs = self._querier.query_e2e_jobs_by_job_type_and_state(
+            from_date, to_date, job_type, job_state
+        )
+        jobs_names = set({})
+        is_top_successful = job_state.value == JobState.SUCCESS.value
+        for job in jobs:
+            if job.name not in jobs_names:
+                jobs_names.add(job.name)
+                job_instance = Job(job.name, self._querier)
+                res.append(
+                    (
+                        job_instance.name,
+                        float(
+                            round(job_instance.get_success_rate(from_date, to_date), 2)
+                        ),
+                    )
+                )
+        res = sorted(res, key=lambda job: job[1], reverse=is_top_successful)
+        return res[0 : min(len(res), n)]
+
+    def _get_top_n_e2e_jobs_triggered(
+        self, from_date: datetime, to_date: datetime, n: int
+    ) -> list[tuple[str, int]]:
+        res = []
+        jobs_names = set({})
+        jobs = self._querier.query_e2e_jobs(from_date, to_date)
+        for job in jobs:
+            if job.name not in jobs_names:
+                jobs_names.add(job.name)
+                job_instance = Job(job.name, self._querier)
+                res.append(
+                    (
+                        job_instance.name,
+                        job_instance.get_number_of_times_triggered(from_date, to_date),
+                    )
+                )
+        res = sorted(res, key=lambda job: job[1], reverse=True)
+        return res[0 : min(len(res), n)]
+
+    def get_report(self, from_date: datetime, to_date: datetime) -> dict[str, Any]:
```

Comment:
I think you should return an object instead of a dict here, that way it will be easier to refactor because the fields will be typed, and it will be able to catch typos and missing fields earlier.